package/MDAnalysis/analysis/contacts.py

```diff
@@ -773,3 +536,325 @@ def plot_qavg(self, filename=None, **kwargs):
+        self.grA, self.grB = grA, grB        
+        refA, refB = refgroup
+
+        # contacts formed in reference
+        r0 = distance_array(refA.positions, refB.positions)
+        self.r0 = r0
+        self.mask = r0 < radius
+
+        self.contact_matrix = []
+        self.timeseries = []
+        self.outfile = outfile
+
+    @staticmethod
+    def load(self, filename):
+        """Load the data file."""
+        records = []
+        with openany(filename) as data:
+            for line in data:
+                if line.startswith('#'):
+                    continue
+                records.append(map(float, line.split()))
+        self.timeseries = np.array(records).T
+        try:
+            self.qavg = np.loadtxt(self.outarray)
+        except IOError as err:
+            if err.errno != errno.ENOENT:
+                raise
+
+    def _single_frame(self):
+        grA, grB, r0, mask = self.grA, self.grB, self.r0, self.mask
+
+        # compute distance array for a frame
+        d = distance_array(grA.positions, grB.positions)
+
+        # select only the contacts that were formed in the reference state
+        # r, r0 are 1D array
+        r, r0 = d[mask], r0[mask]
+
+        if self._method == "cutoff":
+            y = r <= r0
+            y = float(y.sum())/mask.sum()
+        elif self._method == "best-hummer":
+            y, _ = best_hummer_q(r, r0, self.beta, self.lambda_constant)
+        else:
+            raise ValueError("Unknown method type, has to be 'cutoff' or 'best-hummer'")
+
+        cm = np.zeros((grA.positions.shape[0], grB.positions.shape[0]))
+        cm[mask] = y
+        self.contact_matrix.append(cm)
+        self.timeseries.append((self._ts.frame , y, mask.sum()))
+
+    def _conclude(self):
+        """Finalise the timeseries you've gathered.
+
+        Called at the end of the run() method to finish everything up.
+        """
+        # write output
+        if not self.outfile: return
+        with open(self.outfile, "w") as f:
+            f.write("# q1 analysis\n# nref = {0:d}\n".format(self.mask.sum()))
+            f.write("# frame  q1  n1\n")
+            for frame, q1, n1 in self.timeseries:
+                f.write("{frame:4d}  {q1:8.6f} {n1:5d}\n".format(**vars()))            
+
+    def qarray(self, d, out=None):
+        """Return distance array with True for contacts.
+
+        *d* is the matrix of distances. The method uses the value of
+        :attr:`ContactAnalysis1.radius` to determine if a ``distance < radius``
+        is considered a contact.
+
+        If *out* is supplied as a pre-allocated array of the correct
+        shape then it is filled instead of allocating a new one in
+        order to increase performance.
+
+        This method is typically only used internally.
+        """
+        if out:
+            out[:] = (d <= self.radius)
+        else:
+            out = (d <= self.radius)            
+        return out
+
+    def qN(self, q, out=None):
```

Comment:
This is just a copy of an old function that was in ContactAnalysis1, called `qN`. Changing it breaks the API, which is fine by me.